File: fix-md.js
const fs = require("fs");
let c = fs.readFileSync("/root/joblink-v2/public/tom.html", "utf8");

// Replace the broken md function with a working one
const broken = c.indexOf("function md(s)");
const brokenEnd = c.indexOf("}", broken) + 1;
const oldFunc = c.substring(broken, brokenEnd);

const newFunc = `function md(s) {
  if (!s) return "";
  return s
    .replace(/\*\*(.+?)\*\*/g, "<strong>$1</strong>")
    .replace(/\*(.+?)\*/g, "<em>$1</em>")
    .replace(/\x60([^\x60]+)\x60/g, "<code>$1</code>")
    .replace(/^- (.+)/gm, "<li>$1</li>")
    .replace(/(<li>.*<\/li>)/gs, "<ul>$1</ul>")
    .replace(/\n/g, "<br>");
}`;

c = c.substring(0, broken) + newFunc + c.substring(brokenEnd);
fs.writeFileSync("/root/joblink-v2/public/tom.html", c);
console.log("Fixed md function");

File: fix-md.py
import sys

with open(sys.argv[1], "rb") as f:
    data = f.read()

start = data.find(b"function md(s)")
pos = data.find(b"{", start)
depth = 0
i = pos
while i < len(data):
    if data[i:i+1] == b"{":
        depth += 1
    elif data[i:i+1] == b"}":
        depth -= 1
        if depth == 0:
            end = i + 1
            break
    i += 1

new_func = b"function md(s) {\n  if (!s) return '';\n  return s\n    .replace(/\*\*(.+?)\*\*/g, '<strong>$1</strong>')\n    .replace(/\*(.+?)\*/g, '<em>$1</em>')\n    .replace(/`([^`]+)`/g, '<code>$1</code>')\n    .replace(/^- (.+)/gm, '<li>$1</li>')\n    .replace(/(<li>.*<\/li>)/gs, '<ul>$1</ul>')\n    .replace(/\n/g, '<br>');\n}"

data = data[:start] + new_func + data[end:]
with open(sys.argv[1], "wb") as f:
    f.write(data)
print("Fixed")
